crates/context/config/src/client/protocol/evm.rs

```diff
@@ -204,7 +199,16 @@ impl Network {
         method: String,
         args: Vec<u8>,
     ) -> Result<Vec<u8>, EvmError> {
-        let client: ReqwestClient = ClientBuilder::default().http(self.client.parse().unwrap());
+        let signer: PrivateKeySigner = PrivateKeySigner::from_str(&self.secret_key).unwrap();
+        let wallet = EthereumWallet::from(signer);
+
+        let provider =
+            ProviderBuilder::new()
+                .wallet(wallet)
+                .on_http(Url::parse(&self.rpc_url).map_err(|e| EvmError::Custom {
+                    operation: ErrorOperation::Mutate,
+                    reason: e.to_string(),
+                })?);
```

Comment:
move these to `EvmTransport::new`

and store `Provider::builder().wallet(..).on_http(..)` instead of `RpcClient` in the state

Reply:
Updated